nifi-stateless/nifi-stateless-api/src/main/java/org/apache/nifi/stateless/config/PropertiesFileEngineConfigurationParser.java

```diff
@@ -218,4 +223,24 @@ private String getRequired(final Properties properties, final String key) throws
         return propertyValue.trim();
     }
 
+    private String getSensitivePropsKey(final File propertiesFile, final Properties properties) {
```

Comment:
What do you think about pulling some of this logic into a common utility to be used by both `NiFiPropertiesLoader` and this class?

Reply:
I considered that, but also wanted to avoid complicating the dependency tree.  The `nifi-stateless-api` has no dependencies other than `nifi-api`, and should definitely avoid depending on `nifi-properties` or `nifi-properties-loader`. A new module is one option, but it seems a bit much for a single class and method. What do you think?

Reply:
I was thinking something along the lines of `nifi-commons/nifi-security-utils`, but I see now that it has a dependency on `nifi-api`.  I agree that a new module would be a bit heavy-handed in this case, so perhaps a small amount of duplication here is acceptable.

Reply:
Yes, we definitely want to avoid depending on `nifi-security-utils` given the other transitive dependencies included. It is something to keep in mind for the future.